altapay.php

```diff
@@ -1,544 +1,664 @@
+
+		if ( $terminals ) {
+			foreach ( $terminals as $terminal ) {
+				// Load Terminal information
+				$terminalInfo = json_decode( get_option( 'altapay_terminals' ) );
+				$terminalName = $terminal;
+				foreach ( $terminalInfo as $term ) {
+					if ( $term->key === $terminal ) {
+						$terminalName   = $term->name;
+						$terminalNature = $term->nature;
+						if ( in_array( 'CreditCard', $terminalNature ) ) {
+							$tokenStatus = 'CreditCard';
+						}
+					}
+				}
+				// Check if file exists
+				$path    = $terminalDir . $terminal . '.class.php';
+				$tmpPath = $tmpDir . '/' . $terminal . '.class.php';
+
+				if ( file_exists( $path ) ) {
+					require_once $path;
+					$methods[ $terminal ] = 'WC_Gateway_' . $terminal;
+				} elseif ( file_exists( $tmpPath ) ) {
+					require_once $tmpPath;
+					$methods[ $terminal ] = 'WC_Gateway_' . $terminal;
+				} else {
+					// Create file
+					$template = file_get_contents( $pluginDir . 'views/paymentClass.tpl' );
+					$filename = $terminalDir . $terminal . '.class.php';
+					// Check if terminals folder is writable or use tmp as fallback
+					if ( ! is_writable( $terminalDir ) ) {
+						$filename = $tmpDir . '/' . $terminal . '.class.php';
+					}
+					// Replace patterns
+					$content = str_replace( array( '{key}', '{name}', '{tokenStatus}' ), array( $terminal, $terminalName, $tokenStatus ), $template );
+
+					file_put_contents( $filename, $content );
+				}
+			}
+		}
+		return $methods;
+	}
+
+	// Make sure payment page form loads payment template
+	add_filter( 'template_include', 'altapay_page_template', 99 );
+
+	/**
+	 * Load payment template
+	 *
+	 * @param string $template Template to load.
+	 *
+	 * @return string
+	 */
+	function altapay_page_template( $template ) {
+		// Get payment form page id
+		$paymentFormPageID = esc_attr( get_option( 'altapay_payment_page' ) );
+		if ( $paymentFormPageID && is_page( $paymentFormPageID ) ) {
+			// Make sure the template is only loaded from AltaPay.
+			// Load template override
+			$template = locate_template( 'altapay-payment-form.php' );
+
+			// If no template override load template from plugin
+			if ( ! $template ) {
+				$template = dirname( __FILE__ ) . '/views/altapay-payment-form.php';
+			}
+		}
+
+		return $template;
+	}
+
+	// Capture function
+	add_action( 'add_meta_boxes', 'altapayAddMetaBoxes' );
+	add_action( 'wp_ajax_altapay_capture', 'altapayCaptureCallback' );
+	add_action( 'wp_ajax_altapay_refund', 'altapayRefundCallback' );
+	add_action( 'wp_ajax_altapay_release_payment', 'altapayReleasePayment' );
+	add_action( 'admin_footer', 'altapayActionJavascript' );
+	// Create payment page function
+	add_action( 'altapay_checkout_order_review', 'woocommerceOrderReview' );
+	add_action( 'wp_ajax_create_altapay_payment_page', 'createAltapayPaymentPageCallback' );
+
+	/**
+	 * Register meta box for order details page
+	 */
+	function altapayAddMetaBoxes() {
+		global $post, $woocommerce;
```

Comment:
Remove unused global
```suggestion
		global $post;
```